### DISCUSSION TITLE
Select different folders to save images clipped using using tiddlyclip and save media add on

### DISCUSSION THREAD

**Discussion by arunnbabu81** [buggyj/tiddlyclip-plugin#119] [Ideas]
I would like to save the images clipped using tiddlyclip and saved locally using save media add-on in different folders instead of the single `media` folder. Is it possible ? 
In browsers there is an option to automatically download to a specified folder and another option to ask where to save before each download. Can such an option possible in save media add on

**Comment by buggyj** [comment]
It may be possible in used subfolders either now or in a future version -eg 'media/twitter'. I have tried to stop the possibility of automatic saving outside of the media subfolder as this could be a security issue. There is no option in savetiddlers to prompt the user to select a different download folder, I could add this in a future release, but I do not know when I would find the time.

**Reply by arunnbabu81** [reply]
> It may be possible in used subfolders either now or in a future version -eg 'media/twitter'. I have tried to stop the possibility of automatic saving outside of the media subfolder as this could be a security issue. There is no option in savetiddlers to prompt the user to select a different download folder, I could add this in a future release, but I do not know when I would find the time.

What do you meant by it may be possible now ? Is it possible with the current add on ? Hope it arrive soon 😊

**Reply by arunnbabu81** [reply]
> It may be possible in used subfolders either now or in a future version -eg 'media/twitter'. I have tried to stop the possibility of automatic saving outside of the media subfolder as this could be a security issue. There is no option in savetiddlers to prompt the user to select a different download folder, I could add this in a future release, but I do not know when I would find the time.

It would be nice if this configuration option is added to the save media add on. I wanted to organise the clipped images systematically. It would be nice to have this option before large amount of images pile up in my media folder. Especially since I am using relative path to display the images in my wiki. I would have to rename the image file if I change the location later.

**Reply by arunnbabu81** [reply]
@buggyj Is this possible ?

**Reply by buggyj** [reply]
no. I have to be very careful with security.

**Reply by buggyj** [reply]
I have add a new macro to save to the 'files' subdir that can be used instead of `@csaver()` called `@savetofiles()`

**Reply by arunnbabu81** [reply]
> I have add a new macro to save to the 'files' subdir that can be used instead of @csaver() called @savetofiles()
 
Is it a tiddlyclip macro or for the save media add on.
How to upgrade to this version


> There is no option in savetiddlers to prompt the user to select a different download folder, I could add this in a future release, but I do not know when I would find the time.

Can this be an option? I want to save images in different subfolders

**Reply by buggyj** [reply]
@savetofiles() is a tiddlyclip macro.

I will think again about the security, but I am busy at the moment.

**Reply by arunnbabu81** [reply]
> @savetofiles() is a tiddlyclip macro.

What does this mean ? how to use this macro ?
I need some way to change the folder to which the media is getting saved. More advanced saving options might not be needed. Can the filepath of the `file-saving` folder be saved in some config tiddler and this could be changed as needed.

**Reply by buggyj** [reply]
The savetofiles() macro is used instead of csaver() macro - csaver puts the clips into 'media' subdir and savetofiles puts clips into 'files' subdir. - you need 'save media' plugin for this.

**Comment by arunnbabu81** [comment]
Some times I want to save my images in different sub-folders of the `files `folder. Is it possible to add a config tiddler which decide in which subdirectory folder I want to save the images using save media add-on. In this config tiddler, I want define the relative path of the subdirectory folder for saving images.

**Reply by buggyj** [reply]
Allowing relative paths is not something I will support.  I may expand the function of 'savemedia' when I get time to include a second subdir to allow for 'files/subdir1' 'files/subdir2' etc.

**Reply by arunnbabu81** [reply]
> I may expand the function of 'savemedia' when I get time to include a second subdir to allow for 'files/subdir1' 'files/subdir2' etc.

I will be very grateful if you could incorporate this configurable 2nd subdirectory option into the savemedia addon.

**Reply by arunnbabu81** [reply]
@buggyj will this happen any time soon ?

**Reply by buggyj** [reply]
I will look at this when preparing the new release of the tiddlyclip plugin.

**Reply by arunnbabu81** [reply]
@buggyj  just a friendly remainder for this feature request

**Reply by buggyj** [reply]
@arunnbabu81 in June google will be disabling my old Addons, so I need to rewrite savemedia and savetiddlers and will include this change.

**Reply by arunnbabu81** [reply]
> @arunnbabu81 in June google will be disabling my old Addons, so I need to rewrite savemedia and savetiddlers and will include this change.

@buggyj I saw a new test version of the add ons for chrome. Is there a new version available for Firefox?

**Reply by buggyj** [reply]
There is no new version for firefox yet...

**Reply by arunnbabu81** [reply]
> There is no new version for firefox yet...

Will it come out soon ?.....I rarely use chrome because some of my wiki customisations work well only in firefox. So i will definitely need a firefox add on.

**Reply by buggyj** [reply]
At present firefox is saying it will keep support for v2 addons so savemedia etc will keep working.

**Reply by arunnbabu81** [reply]
> At present firefox is saying it will keep support for v2 addons so savemedia etc will keep working.

> @arunnbabu81 in June google will be disabling my old Addons, so I need to rewrite savemedia and savetiddlers and will include this change.

But firefox version won't support the feature request in this post rite

**Reply by buggyj** [reply]
That is correct. I had hoped that the new v3 versions of my plugins would work on firefox - but they did not work (on firefox, it appears that mozilla still have a lot of work to do for the v3 addon support). I will make a separate update to the v2 version of savemedia for firefox.

**Reply by arunnbabu81** [reply]
@buggyj For the time being, I will use chrome or edge browser. Is there an updated version of the tiddlyclip plug-in which works with the new savemedia and tiddlyclip addons. Can you provide the plug in link ? I did extracted the new add-ons from the github page and installed them in my chrome/edge browsers.
Can you tell me how to change the folder used for saving images using savemedia add-on

**Reply by arunnbabu81** [reply]
> Can you tell me how to change the folder used for saving images using savemedia add-on

Is there any way to do it ?
Also is there an updated tiddlyclip  plug in ?

**Reply by buggyj** [reply]
@arunnbabu81 Hi arunnbabu, I have a  lot of projects on the go, and have not had much time to look at this. I now have the changes for the update to tiddlyclip plugin, but I am in the middle of changing systems and so will not commit them for a few days.

**Reply by buggyj** [reply]
@arunnbabu81 there is now a new pre-release at https://bjhacks.tiddlyhost.com/ with the new command. Use it like this:

`((*@saveto(@pageTitle,@snap,mydir)*))`

**Reply by arunnbabu81** [reply]
Does this work only with the new add-on for chrome?

**Reply by buggyj** [reply]
It should work on firefox as well

**Reply by arunnbabu81** [reply]
> @arunnbabu81 there is now a new pre-release at https://bjhacks.tiddlyhost.com/ with the new command. Use it like this:
> 
> `((*@saveto(@pageTitle,@snap,mydir)*))`

How to modify my code if I want to save the image in a subdirectory (for example msk-rad) of the files or media folder
This is the code I am currently using 
```
#rule=/^(.*?).pdf$/g
#term=$1
@text=((*@removelf(@text)*))
#datetime=((*@now([UTC]YYYY0MM0DD0hh0mm0ssXXX)*))
#ext=png
#title=((*@section/#rule/#term*))((*#datetime*)).((*#ext*))
#astr=((*#title*))
#match=/(\s)/g
#subst=-
#astr=((*#astr/#match/#subst*))
$text=((*$text*))![](./((*@csaver(#astr,@clipsnap)*)))\n
```

How to modify it ?

**Reply by buggyj** [reply]
That is not possible. The new saver replaces `((*@csaver(#astr,@clipsnap)*)) `with `((*@saveto(@pageTitle,@snap,mydir)*))`. It is only possible to give a directory name, not a path. So you can only choose other names for the directory e.g. "files", "photos", "mydir" 

You can use the directory name directly as in the example or you can use a tiddlyclip variable:
```

#mediadir="mydir"
$text=((*$text*))![](./((*@saveto(#astr,@clipsnap,#mediadir)*)))\n
```

**Reply by arunnbabu81** [reply]
> That is not possible. The new saver replaces `((*@csaver(#astr,@clipsnap)*)) `with `((*@saveto(@pageTitle,@snap,mydir)*))`. It is only possible to give a directory name, not a path. So you can only choose other names for the directory e.g. "files", "photos", "mydir" 
> 
> You can use the directory name directly as in the example or you can use a tiddlyclip variable:
> ```
> 
> #mediadir="mydir"
> $text=((*$text*))![](./((*@saveto(#astr,@clipsnap,#mediadir)*)))\n
> ```

So subdirectories are not supported in this version ?

**Reply by buggyj** [reply]
Adding paths would make the software a lot easier to hack. There will be many AI's looking for software on github that access the file system, and then they will try to find exploits and automatic generate code to hack computers.

**Comment by arunnbabu81** [comment]
@buggyj do you know of a way to transfer files from one folder to another automatically ?

**Reply by buggyj** [reply]
for what operating system??

In linux there are options also symbolic links can be used

**Reply by arunnbabu81** [reply]
> for what operating system??
> 
> In linux there are options also symbolic links can be used

For windows

**Reply by buggyj** [reply]
I am not a windows user
